Infrastructure/gpg-ipdeny/README.md

```diff
@@ -0,0 +1,44 @@
+# gpg-ipdeny
+
+gpg-ipdeny is an IP denylisting nginx application based upon the IP allowing version found here: [GOV.UK PaaS IP authentication route service](https://github.com/alphagov/paas-ip-authentication-route-service)
+
+It is designed to allow access to the Gender Pay Gap Service to all IP addresses except those specifically set, returning a 403 response to those banned IPs.
+
+## Deployment
+
+Since the list of banned IPs should not update particularly frequently, the application is deployed manually via the Deploy.sh script.
+
+The script relies up having the command line processor [jq](https://stedolan.github.io/jq/) available. 
+
+When using gitbash for Windows this can simply be added by running the following command, it should also be possible to add it to the system path:
+```
+curl -L -o /usr/bin/jq.exe https://github.com/stedolan/jq/releases/latest/download/jq-win64.exe
+```
+
+Before running the deployment script you must be logged into GOV.UK PaaS using the Cloud Foundry CLI.
+
+The deployment script takes the following parameters:
+* -a PROTECTED_APP_NAME The application to protect
+* -e PROTECTED_APP_SPACE_NAME The name of the space the apps are in
+* -i DENIED_IPS A comma separated list of IP addresses to be banned
+* -r ROUTE_SERVICE_APP_NAME The name of the service app to create/update
+* -s ROUTE_SERVICE_NAME The name of the service to create/update
+
+For example to set up ip denylisting on the dev space the command could be run (using the gitbash window set up with jq earlier) as 
+```
+./deploy.sh -e gpg-dev -r gpg-dev-ipdeny -s gpg-dev-ipdeny-service -a gender-pay-gap-dev -i "1.2.3.4,2.3.4.5"
+```
+
+The comma separated list of banned IPs can be found in Zoho, when updating this list, newlines should be avoided as they can cause syntax errors within the nginx.conf. This can be seen by an error when deploying, and then if you try and run
```

Comment:
The list in Zoho is quite long and I can see it being quite hard to maintain. With that in mind I wonder how much extra work it would be to use a file here - e.g.
* Argument is a filename
* File is basically "exactly what you found in Zoho" (i.e. what's in Zoho would need editing to remove comments)
* File comprises one entry per line, meaning it's easier to read/maintain
* Script replaces newlines with commas

thoughts?

Reply:
Agreed that it would be better to do something like that, would just need to figure out how